kasa/smartstrip.py

```diff
@@ -179,13 +185,20 @@ async def get_emeter_monthly(self, year: int = None, kwh: bool = True) -> Dict:
         :param year: year for which to retrieve statistics (default: this year)
         :param kwh: return usage in kWh (default: True)
         """
-        emeter_monthly: DefaultDict[int, float] = defaultdict(lambda: 0.0)
-        for plug in self.children:
-            plug_emeter_monthly = await plug.get_emeter_monthly(year=year, kwh=kwh)
-            for month, value in plug_emeter_monthly:
-                emeter_monthly[month] += value
+        return await self._async_get_emeter_sum(
+            "get_emeter_monthly", {"year": year, "kwh": kwh}
+        )
 
-        return emeter_monthly
+    async def _async_get_emeter_sum(self, func: str, kwargs: Dict[str, Any]) -> Dict:
+        """Retreive emeter stats for a time period from children."""
+        self._async_verify_emeter()
+
+        emeter: DefaultDict[int, float] = defaultdict(lambda: 0.0)
+        for plug in self.children:
+            emeter_daily = await getattr(plug, func)(**kwargs)
+            for day, value in emeter_daily.items():
+                emeter[day] += value
+        return emeter
```

Comment:
Instead of doing this, another maybe better option would be making `update()` on child to do perform the emeter queries (see https://github.com/python-kasa/python-kasa/blob/master/kasa/smartdevice.py#L466) to update their cache. This way the property-based API for emeter would work like for plugs (i.e., require no I/O outside the update call).

Reply:
done